### PR TITLE
Remove state from state viewer

### PR DESCRIPTION
**Pull request by bowenwang1996**

### CODE REVIEW

File: node/http/src/api.rs [hunk 1]
```diff
@@ -62,11 +62,15 @@ impl HttpApi {
             r.contract_account_id,
             r.method_name,
         );
-        let best_block = self.client.shard_chain.chain.best_block();
+        let state_update = self.client.shard_chain.get_state_update();
```
Comment (azban) [inline node/http/src/api.rs]: we are continuing to add more logic like this to the network interface layer .. this kind of stuff should be wrapped by the client cc: @nearmax

File: node/runtime/src/state_viewer.rs [hunk 2]
```diff
@@ -19,9 +18,7 @@ pub struct ViewStateResult {
     pub values: HashMap<Vec<u8>, Vec<u8>>
 }
 
-pub struct StateDbViewer {
-    state_db: Arc<StateDb>,
-}
+pub struct StateDbViewer {}
```
Comment (azban) [inline node/runtime/src/state_viewer.rs]: this should just be a module if we aren't going to keep any struct members

Comment (bowenwang1996) [reply node/runtime/src/state_viewer.rs]: Yeah. Will change that later

Comment (azban) [reply node/runtime/src/state_viewer.rs]: let's make an issue then

File: node/http/src/api.rs (hunk 1 above)
Comment (nearmax) [reply node/http/src/api.rs]: Actually `state_update` is something that client will eventually expose, but you are right a lot of stuff will be wrapped into the client.

Comment (azban) [reply node/http/src/api.rs]: https://github.com/nearprotocol/nearcore/issues/536

File: node/runtime/src/state_viewer.rs (hunk 2 above)
Comment (ilblackdragon) [reply node/runtime/src/state_viewer.rs]: Actually what do you think of practice  having structs with no fields just to group methods?

Comment (bowenwang1996) [reply node/runtime/src/state_viewer.rs]: I am ambivalent about it. Not sure which one is more idiomatic

Comment (azban) [reply node/runtime/src/state_viewer.rs]: pretty sure that's the point of a file or module

File: node/http/src/api.rs (hunk 1 above)
Comment (ilblackdragon) [reply node/http/src/api.rs]: It's not really a client if it's not needed for operation of the protocol IMO. A view later that is shared between different accessors like RPC, we sockets etc is fine.

Comment (azban) [reply node/http/src/api.rs]: yea.. we've hinted on that here https://github.com/nearprotocol/nearcore/issues/479

maybe #536 is a dupe

Comment (ilblackdragon) [reply node/http/src/api.rs]: A reasoning is the client will need to be triple reviewed / bug bounty and etc vs extra code that is not involved in operation of protocol can be more relaxed.

Comment (azban) [reply node/http/src/api.rs]: > A reasoning is the client will need to be triple reviewed / bug bounty and etc vs extra code that is not involved in operation of protocol can be more relaxed.

yea, feel like that was the motivation behind #479

Comment (nearmax) [reply node/http/src/api.rs]: I like the idea of separating view layer from the domain logic, if the only thing view layer does is providing convenience accessors to stuff inside the client, and it does not mutate client's state.

File: node/runtime/src/state_viewer.rs (hunk 2 above)
Comment (nearmax) [reply node/runtime/src/state_viewer.rs]: Empty structs are a normal thing in Rust, one can use them in combination with traits. Suppose we had two different ways to implement `StateDbViewer` functionality (e.g. DebugNet and TestNet), then we could create a common trait and make RPC depend on that trait. On the other hand, if had two versions of `StateDbViewer` implemented as two different modules then we would not be able to have RPC that can easily switch between them.

In the current situation, however, we do not have alternative implementations of `StateDbViewer` so it probably does not matter whether it is a module or an empty struct.

**Pull request merged**